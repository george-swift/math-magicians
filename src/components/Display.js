import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="col-12 h-25 text-white">
    <div className="problem">10 x 0</div>
    <div className="result">
      {result}
    </div>
  </div>
);

Display.propTypes = {
  result: PropTypes.number,
};

Display.defaultProps = {
  result: 0,
};

export default Display;
