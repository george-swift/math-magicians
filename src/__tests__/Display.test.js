import renderer from 'react-test-renderer';
import Display from '../components/Display';

test('Display renders outcomes of calculations correctly', () => {
  const tree = renderer
    .create(<Display result="10" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Display has default prop value of 0', () => {
  const tree = renderer
    .create(<Display />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Display renders operation correctly', () => {
  const tree = renderer
    .create(<Display operation="x" result="10" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
