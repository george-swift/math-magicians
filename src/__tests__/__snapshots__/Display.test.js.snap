// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display has default prop value of 0 1`] = `
<div
  className="col-12 h-25 text-white"
>
  <div
    className="problem"
  />
  <div
    className="result"
  >
    0
  </div>
</div>
`;

exports[`Display renders outcomes of calculations correctly 1`] = `
<div
  className="col-12 h-25 text-white"
>
  <div
    className="problem"
  />
  <div
    className="result"
  >
    10
  </div>
</div>
`;
